game/scripts/vscripts/Sections/Chapter3/Chapter3.ts

```diff
@@ -111,279 +43,199 @@ const onStart = (complete: () => void) => {
         Vector(-2608, -648, 265),
         100
-    )!;
-    const neutralCampPostion = neutralCamp?.GetAbsOrigin();
+    ), "Could not find neutral camp spawner");
+    const neutralCampPostion = neutralCamp.GetAbsOrigin();
 
     GameRules.SpawnNeutralCreeps();
-    const neutralCamps = Entities.FindAllByClassname(
-        "npc_dota_neutral_spawner"
-    );
-
-    graph = tg.forkAny([
-        tg.trackGoals(getGoals),
-        tg.seq([
-            tg.wait(1),
-            // Show message and explain what the "neutral spawn indicators" on the minimap are.
-            // Kill them after so that we can make our own.
-            tg.immediate(() => DestroyNeutrals()),
-
-
-            tg.immediate(
-                (context) =>
-                    (context[NeutralGoalKeys.MoveToCamp] = GoalState.Started)
-            ),
-            tg.goToLocation(markerLocation),
-            tg.immediate((context) => {
-                context[NeutralGoalKeys.MoveToCamp] = GoalState.Completed;
-                context[NeutralGoalKeys.UpgradeAbility] = GoalState.Started;
-            }),
-
-            // Tell the player to skill dragon blood for extra armor, so that it's better against attacks (big part of fighting neutrals is not taking too much damage)
-            tg.immediate((context) => {
-                if (playerHero.GetLevel() < 2) {
-                    playerHero.HeroLevelUp(false);
+
+    graph = tg.withGoals(_ => goalTracker.getGoals(), tg.seq([
+        tg.wait(1),
+        // Show message and explain what the "neutral spawn indicators" on the minimap are.
+        // Kill them after so that we can make our own.
+        tg.immediate(_ => DestroyNeutrals()),
+
+        tg.immediate(_ => goalMoveToCamp.start()),
+        tg.goToLocation(markerLocation),
+        tg.immediate(_ => {
+            goalMoveToCamp.complete();
+            goalKillSatyrs.start();
+        }),
+
+        // Create the units
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_trickster",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "trickster_1"
+        ),
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_trickster",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "trickster_2"
+        ),
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_soulstealer",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "soulstealer_1"
+        ),
+        tg.spawnUnit(
+            "npc_dota_neutral_satyr_soulstealer",
+            neutralCampPostion.__add(RandomVector(50)),
+            DotaTeam.NEUTRALS,
+            "soulstealer_2"
+        ),
+
+        tg.immediate((ctx) => {
+            units.push(ctx["trickster_1"] as CDOTA_BaseNPC);
+            units.push(ctx["trickster_2"] as CDOTA_BaseNPC);
+            units.push(ctx["soulstealer_1"] as CDOTA_BaseNPC);
```

Comment:
Is it intended that "soulstealer_2" is not included?

Reply:
I didn't make the code but it looks like we are using that array to check whether these three are dead in a fork. There's another step in the fork that checks whether `soulstealer_2` is dead and creates an item.